src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -240,25 +374,92 @@ inline DocumentStatisticsAnnotationData::DocumentStatisticsAnnotationData(Freque
 
 // ----------------------------------------------------------------------
 // |
-// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |  DocumentStatisticsEstimator
 // |
 // ----------------------------------------------------------------------
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(void) :
-    _totalNumDocuments(0) {
+//Would it be better to place default values during pipeline chaining(countvectorizer and tfidfvectorizer)? if so this ctor will show no default values
+template <size_t MaxNumTrainingItemsV>
+DocumentStatisticsEstimator<MaxNumTrainingItemsV>::DocumentStatisticsEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex,
+                                                                               StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                               nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, std::float_t maxDf) :
+    BaseType(
+        std::move(pAllColumnAnnotations),
+        std::move(colIndex),
+        true,
+        std::move(decorator),
+        std::move(vocabulary),
+        std::move(maxFeatures),
+        std::move(minDf),
+        std::move(maxDf)
+    ) {
 }
 
-inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator) :
+// ----------------------------------------------------------------------
+// |
+// |  Details::DocumentStatisticsTrainingOnlyPolicy
+// |
+// ----------------------------------------------------------------------
+inline Details::DocumentStatisticsTrainingOnlyPolicy::DocumentStatisticsTrainingOnlyPolicy(StringDecorator decorator, nonstd::optional<IndexMap> vocabulary, 
+                                                                                           nonstd::optional<std::uint32_t> maxFeatures, std::float_t minDf, 
+                                                                                           std::float_t maxDf) :
     _stringDecoratorFunc(
         [&decorator](void) -> StringDecorator & {
-            if(!decorator)
-                throw std::invalid_argument("decorator");
-
+            //temporary mute this validate
```

Comment:
Should this code be restored?

Reply:
Not at this moment, because decorator will be generated from strings.h which is not finished yet. It will be restored as decorators are ready in next PR

Reply:
I don't follow. I believe that a decorator is required from the class's perspective? If so, the caller should be the one providing the temporary implementation now and the one in strings.h later.